pennylane/measurements/shots.py

```diff
@@ -154,6 +155,15 @@ def __init__(self, shots):
 
         self._frozen = True
 
+    def __deepcopy__(self, memo):
+        cls = self.__class__
+        shots = self.shot_vector if self.has_partitioned_shots else self.total_shots
+        result = cls.__new__(cls, shots)
+        memo[id(self)] = result
+        for k, v in self.__dict__.items():
+            setattr(result, k, copy.deepcopy(v, memo))
+        return result
+
```

Comment:
Does just 
```suggestion
    def __deepcopy__(self, memo):
        return self
```

Shots is fully immutable, so we should be able to just do this.